flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/aggfunctions/SumAggFunction.scala

```diff
@@ -49,13 +50,25 @@ abstract class SumAggFunction[T: Numeric] extends AggregateFunction[T] {
       val v = value.asInstanceOf[T]
       val a = accumulator.asInstanceOf[SumAccumulator[T]]
       a.f0 = numeric.plus(v, a.f0)
-      a.f1 = true
+      a.f1 += 1
+    }
+  }
+
+  override def retract(accumulator: Accumulator, value: Any): Unit = {
+    if (value != null) {
+      val v = value.asInstanceOf[T]
+      val a = accumulator.asInstanceOf[SumAccumulator[T]]
+      a.f0 = numeric.plus(v, a.f0)
```

Comment:
Added the test to compare two accumulators instead of just comparing two values will catch this.